### PR TITLE
Gh16344 import modular contract log

### PR DESCRIPTION
**Pull request by chemsnouioua**
#16344

### CODE REVIEW

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/impexp/ModCntrLogImportTableSqlUpdater.java [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+				+ selection.toSqlWhereClause("i");
+
+		DB.executeUpdateAndThrowExceptionOnFail(sql, ITrx.TRXNAME_ThreadInherited);
+	}
+	private void dbUpdateContractModule(@NonNull final ImportRecordsSelection selection)
+	{
+		final String sqlContractModuleId = "SELECT c." + I_ModCntr_Module.COLUMNNAME_ModCntr_Module_ID
+				+ " FROM " + I_ModCntr_Module.Table_Name + " c"
+				+ " WHERE c." + I_ModCntr_Module.COLUMNNAME_Name + " = i." + COLUMNNAME_ContractModuleName
+				+ " AND c." + COLUMNNAME_AD_Org_ID + " IN (i." + COLUMNNAME_AD_Org_ID + ", 0)"
```
Comment (adi-stefan) [inline backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/impexp/ModCntrLogImportTableSqlUpdater.java:125]: This is EXTREMELY optimistic. The modules are not required to have an unique name, in fact they will be duplicated if a contract is copied to a new year. you need to also involve the product/harvest year/calendar/sotrx and ideally BP.
Similar to how it's done in `de.metas.contracts.modular.settings.ModularContractSettingsDAO#isSettingsExist`

**Review by adi-stefan — CHANGES_REQUESTED** [review]
please see my comments

File: backend/de.metas.contracts/src/main/sql/postgresql/system/50-de.metas.contracts/5705900_sys_gh16344_import_format_adjustments.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+ALTER TABLE i_modcntr_log
+    DROP COLUMN datetrx
+;
+
```
[the remark this answers is not in the document]
Comment (chemsnouioua) [reply backend/de.metas.contracts/src/main/sql/postgresql/system/50-de.metas.contracts/5705900_sys_gh16344_import_format_adjustments.sql:4]: Dropped the column and recreate it, kept the changes,  i didn't messed up with the scripts.

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/impexp/ModCntrLogImportTableSqlUpdater.java (hunk 1 above)
Comment (chemsnouioua) [reply backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/impexp/ModCntrLogImportTableSqlUpdater.java:127]: <details>
<summary>Final update SQL</summary>

image
</details>

**Review by adi-stefan — APPROVED** [review]

**Pull request merged**